src/components/heatmap/heatmap.tsx

```diff
@@ -0,0 +1,344 @@
+          }))
+        ref.read(1000, header).then(function getPage(page: any) {
+          readAll(page);
+        }).catch(function () {
+          setInvalidConfigs(['Error while reading data from Ermrestjs']);
+          setConfigErrorsPresent(true);
+        });
+      }
+    });
+
+  }, []);
+
+  useEffect(() => {
+    setFacet({
+      'and': [{
+        'markdown_name': 'Gene NCBI ID',
+        'entity': false,
+        'choices': [NCBIGeneID],
+        'source': 'NCBI_GeneID',
+        'hide_not_null_choice': true,
+        'hide_null_choice': true
+      }]
+    });
+  }, [NCBIGeneID]);
+
+
+  const verifyConfiguration = (reference: any) => {
+    const columns = ['titleColumn', 'idColumn', 'xColumn', 'yColumn', 'zColumn'];
+    setInvalidConfigs(() => []);
+    for (let i = 0; i < columns.length; i++) {
+      try {
+        reference.getColumnByName(config.data[columns[i]]);
+      } catch (error) {
+        setInvalidConfigs((prevInvalidConfigs) =>
+          ([...prevInvalidConfigs, 'Coulmn \'' + config.data[columns[i]] + '\' does not exist. Give a valid value for the ' + columns[i] + '.']))
+      }
+    }
+    const sortColumns = config.data.sortBy;
+    for (let i = 0; i < sortColumns.length; i++) {
+      try {
+        reference.getColumnByName(sortColumns[i].column);
+      } catch (error) {
+        setInvalidConfigs((prevInvalidConfigs) =>
+          ([...prevInvalidConfigs, 'Coulmn \'' + sortColumns[i].column + '\' in \'sortBy\' field does not exist. Replace it with a valid column.']));
+      }
+    }
+    if (invalidConfigs.length > 0) {
+      setInvalidConfigs(() => invalidConfigs);
+      setConfigErrorsPresent(true);
+    }
+  }
+
+  /**
+   * @param page : Page object fecthed from response of ermrestURI
+   */
+  const readAll = (page: any) => {
+    for (let i = 0; i < page.tuples.length; i++) {
+      addData(page.tuples[i]);
+    }
+    if (page.hasNext) {
+      Object(header).action = 'page';
+      page.next.read(1000, header).then(readAll).catch(function () {
+        setInvalidConfigs(() => ['Error while reading data from Ermrestjs']);
+        setConfigErrorsPresent(() => false);
+      });
+    } else {
+      setHeatmaps(() => (heatmaps));
+    }
+    setNCBIGeneID(() => (page.tuples[0].data.NCBI_GeneID));
+  }
+  /**
+   * 
+   * @param tuple : Tuple object retrieved from page
+   * Adds data(x,y,z values) to heatmap objects
+   */
+
+  const addData = (tuple: any) => {
+    const configData = Object.assign({}, config.data);
+    let hm = null;
+    const x = tuple.data[configData.xColumn];
+    const y = tuple.data[configData.yColumn];
+    const z = tuple.data[configData.zColumn];
+    const title = tuple.data[configData.titleColumn];
+    const id = tuple.data[configData.idColumn];
+    for (let i = 0; i < heatmaps.length; i++) {
+      if (Object(heatmaps[i]).title === title) {
+        Object(heatmaps)[i]['id'] = id;
+        hm = heatmaps[i];
+      }
+    }
+    if (hm === null) {
```

Comment:
Can you add a comment explaining why this check is being done in conjunction with the above for loop?